File: MATLAB/Factorial.m
flag = 1;
num = 0;

while flag == 1

    try
        
    archivo = fopen('M_registro.txt','at');
    
    fprintf('\n');
	fprintf('       CALCULADORA DE FACTORIAL       \n');
	fprintf('\n');
	fprintf('######################################\n');
	fprintf('# 1. CALCULAR EL FACTORIAL ----- (f) #\n');
	fprintf('# 2. SALIR ----------------------(s) #\n'); 
	fprintf('######################################\n');
    
    fprintf('\n');
    opt = input('ELIJA UNA OPCIÓN: ','s');
    fprintf('\n');
    fact = 1;
    
    switch opt
        
        case 'f'
            
            num = input('INGRESE UN NUMERO ENTERO POSITIVO: ');
            
            if num == 0
                fact = 1;
                fprintf('EL FACTORIAL DE %g ES: %g \n', num, fact);
                fprintf(archivo, 'EL FACTORIAL DE %g ES: %g\n',num, fact);
                fclose(archivo);
                
            else
                
            if num > 0
                 
                for i = 1:num
                fact = fact*i;
                end
                
                fprintf('EL FACTORIAL DE %g ES: %g \n', num, fact);
                fprintf(archivo, 'EL FACTORIAL DE %g ES: %g\n',num, fact);
                fclose(archivo);
      
            else
                
                opt = w;
                
            end                               
            end
            
        
        case 's'
            
            flag = 0;
        
        otherwise
            
            fprintf('######################################\n');
			fprintf('# USTED NO INGRESO UNA OPCIÓN VALIDA #\n');
			fprintf('######################################\n');
 
    end
    
    catch
        fprintf('\n');
        fprintf('######################################\n');
		fprintf('# ¡ERROR! INGRESE UN ENTERO POSITIVO #\n');
		fprintf('######################################\n');
    end
        
end

fprintf('####################################\n');
fprintf('# GRACIAS POR UTILIZAR EL PROGRAMA #\n');
fprintf('####################################\n');

File: PYTHON/factorial.py
flag = True
fact = int(1)

while flag == True:

	
	try:
		
		archivo = open("P_registro.txt","a")
		print("")
		print("       CALCULADORA DE FACTORIAL       ")
		print("")
		print("######################################")
		print("# 1. CALCULAR EL FACTORIAL ----- (f) #")
		print("# 2. SALIR ----------------------(s) #") 
		print("######################################")
		print("")

		opt = input("INGRESE LA OPCIÓN QUE ELIGIÓ: ")
		print("")

		if opt == "f":

			num = int(input("INGRESE EL NÚMERO ENTERO AL CUAL DESEA CALCULAR EL FACTORIAL: "))

			if num == 0:

				fact=1

				print("")
				print("EL FACTORIAL DE ", num, " ES: ", fact)
				archivo.write('EL FACTORIAL DE %s'%num)
				archivo.write(' ES: %s'%fact +'\n')
				archivo.close()

			elif num > 0:

				fact = 1

				for i in range(1,num+1):

					fact *= i 

				print("")
				print(" EL FACTORIAL DE ", num, " ES: ", fact)
				archivo.write('EL FACTORIAL DE %s'%num)
				archivo.write(' ES: %s'%fact +'\n')
				archivo.close()

			else:

				fact = s

		elif opt == "s":

			flag = False

		else: 

			
			print("######################################")
			print("# USTED NO INGRESO UNA OPCIÓN VALIDA #")
			print("######################################")

	except:

		print("")
		print("#######################################")
		print("# ¡INGRESE UN NÚMERO ENTERO POSITIVO! #")
		print("#######################################")

print("#####################################")
print("# GRACICAS POR UTILIZAR EL PROGRAMA #")
print("#####################################")